### PR TITLE
feat(purge): add `--include-locked` flag to purge command

### PR DESCRIPTION
**Pull request by balcsida**
We lock all the tags (except `latest`) by default to prevent any overwrite. However, we would like to some purging here and there.
So, with the help of Claude, I wrote these changes which we currently use. This introduces `--include-locked` flag to purge command, which will annotate the images that have the tags writeEnabled or deleteEnabled set to false.
If this flag is used without `--dry`, ACR CLI will attempt to release these locks and then do the purge on the tag/manifest.
This PR also includes a little change to make the output of the dry run a little less scarier.

### CODE REVIEW

File: internal/worker/purger.go [hunk 1]
```diff
@@ -70,22 +92,45 @@ func (p *Purger) PurgeManifests(ctx context.Context, manifests []string) (int, e
 	group := p.pool.NewGroup()
 	for _, manifest := range manifests {
 		group.SubmitErr(func() error {
+			// If include-locked is enabled, we need to check if manifest is locked and unlock it first
+			if p.includeLocked {
+				manifestAttrs, err := p.acrClient.GetAcrManifestAttributes(ctx, p.repoName, manifest)
```
Comment (estebanreyl) [inline internal/worker/purger.go:97]: It seems to me like we are doing a lot of extra work here. GetManifestAttributes for every single image will slow down the purge dramatically since it results in one extra Get attributes call per manifest delete (I can run it thorugh validation checks to see exactly whats the expense but I am confident for large repositories this is going to be quite time consuming), instead, my suggestion would be to modify the manifests [] string input to be one that will include the information about if the manifest is locked since this information is already known in the exploratory phase.

File: internal/worker/purger.go [hunk 2]
```diff
@@ -70,22 +92,45 @@ func (p *Purger) PurgeManifests(ctx context.Context, manifests []string) (int, e
...
+						
+						enabledTrue := true
+						unlockAttrs := &acr.ChangeableAttributes{
+							DeleteEnabled: &enabledTrue,
+							WriteEnabled:  &enabledTrue,
+						}
+						
+						_, unlockErr := p.acrClient.UpdateAcrManifestAttributes(ctx, p.repoName, manifest, unlockAttrs)
+						if unlockErr != nil {
+							fmt.Printf("Failed to unlock %s/%s@%s, error: %v\n", p.loginURL, p.repoName, manifest, unlockErr)
```
Comment (estebanreyl) [inline internal/worker/purger.go:110]: Similar to above, does this really constitute a failure we cannot ignore?

File: README.md [hunk 3]
```diff
@@ -208,6 +208,16 @@ acr purge \
     --repository-page-size 10
 ```
 
+#### Include-locked flag
```
Comment (estebanreyl) [inline README.md:211]: Can we add a warning to this? Maybe a link to https://learn.microsoft.com/en-us/azure/container-registry/container-registry-image-lock to note that this is the lock types that are being referenced?

File: cmd/acr/annotate.go [hunk 4]
```diff
@@ -288,9 +309,9 @@ func annotateUntaggedManifests(ctx context.Context,
 	// Contrary to getTagsToAnnotate, getManifests gets all the manifests at once.
 	// This was done because if there is a manifest that has no tag but is referenced by a multiarch manifest that has tags then it
 	// should not be annotated.
-	manifestsToAnnotate, err := common.GetUntaggedManifests(ctx, poolSize, acrClient, repoName, true, nil, dryRun)
+	manifestsToAnnotate, err := common.GetUntaggedManifests(ctx, poolSize, acrClient, repoName, true, nil, dryRun, includeLocked)
 	if err != nil {
-		return -1, err
+		return -1, 0, err
```
Comment (estebanreyl) [inline cmd/acr/annotate.go:314]: It seems like the skipped value is always 0 for this one? Is there a reason to add this return?

**Comment by balcsida** [comment]
Thank you so much for taking the time to check my PR so thoroughly 🙇
I'm going to try to address all your review findings.

File: cmd/acr/annotate.go (hunk 4 above)
Comment (balcsida) [reply cmd/acr/annotate.go]: Yup, you are correct, my mistake

**Comment by estebanreyl** [comment]
Just took a look at the latest updates and it looks good to me. I'd like to run it once against a real registry first before merging and will setup a quick scenario tomorrow, but I have no real concerns at this moment. Thanks for the contributions :)

**Comment by balcsida** [comment]
I would like to thank you so much for checking these! Let me create some helper scripts to generate test repo with a bunch of images.

I've used a binary built from an earlier commit to clean up ~24k tags and manifests, it was much faster than generating a list of Azure CLI commands to run an then running them. 😅

File: scripts/README-purge-testing.md [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+- Basic purge functionality and dry-run verification
+- Lock testing (all combinations of writeEnabled/deleteEnabled)
+- Pattern matching and regex support
+- Keep parameter functionality
+- Concurrent operations with varying worker counts
+- Manifest vs tag deletion scenarios
+- Age-based filtering
+- Edge cases and error handling
+
+### 2. generate-test-images.sh
```
Comment (estebanreyl) [inline scripts/README-purge-testing.md:65]: I am guessing copilot was used here and it generated info for a script no longer there?

File: scripts/test-purge-all.sh [hunk 6]
```diff
@@ -0,0 +1,1330 @@
+#!/bin/bash
```
Comment (estebanreyl) [inline scripts/test-purge-all.sh:1]: Having the script is great but am not sure to what extent I'd like to consider it fully reliable yet. Can we add it to an /experimental folder under scripts?

File: scripts/README-purge-testing.md (hunk 5 above)
Comment (balcsida) [reply scripts/README-purge-testing.md:65]: I used Claude Code for this PR, but this was a human error from my side 😅

**Review by estebanreyl — APPROVED** [review]
The scripts are a great addition; I went over them and ran them as well and it's a valuable tool for testing. The other changes are also fantastic. Thank a lot for the great contribution! LGTM at this point

**Comment by balcsida** [comment]
Thank you again and no worries for the delay!
I caused some lint errors, I sent PR #483 to fix them
